### PR TITLE
chore: enhance .NET onboarding story

### LINKED ISSUE #1730: [Question]: Can I provide a Chromium channel when using PageTest in unit?
### Your question

Hello,

I can specify BROWSER and HEADED through the means of environment variables, to be used with a PageTest base class in my code. I've looked through the playwright sources, and was unable to find an environment variable for a Chromium channel. Is it possible to somehow control it in this scenario (for example, by setting some property, etc.)?

### PR DESCRIPTION
**Pull request by mxschmitt**
Fixes https://github.com/microsoft/playwright-dotnet/issues/1730

Preview https://delightful-forest-0a29f6210-654.centralus.azurestaticapps.net/

Aligned the .NET onboarding story with debbie to Node.js

### CODE REVIEW

File: docs/src/browsers.md [hunk 1]
```diff
@@ -157,6 +157,19 @@ See all supported browsers:
 pwsh bin\Debug\netX\playwright.ps1 install --help
 ```
 
+## Install browsers via API
+* langs: csharp
+
+It's possible to run [Command line tools](./cli.md) commands via the .NET API:
+
+```csharp
+var exitCode = Microsoft.Playwright.Program.Main(new[] {"install"});
```
Comment (dgozman) [inline docs/src/browsers.md:166]: Is this an official way to run our program? Do we guarantee it's not going to change?

File: docs/src/library-csharp.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+Now run it.
+
+```bash
+dotnet run
+```
+
+By default, Playwright runs the browsers in headless mode. To see the browser UI, pass the `Headless = false` flag while launching the browser. You can also use [`option: slowMo`] to slow down execution. Learn more in the debugging tools [section](./debug.md).
+
+```csharp
+await playwright.Firefox.LaunchAsync(new()
```
Comment (dgozman) [inline docs/src/library-csharp.md:49]:
```suggestion
await using var browser = await playwright.Firefox.LaunchAsync(new()
```

File: docs/src/test-runners-csharp.md [hunk 3]
```diff
@@ -108,6 +108,69 @@ By default NUnit will run all test files in parallel, while running tests inside
 
 For CPU-bound tests, we recommend using as many workers as there are cores on your system, divided by 2. For IO-bound tests you can use as many workers as you have cores.
 
+### Customizing [BrowserContext] options
+
+To override the context options, you can override the `ContextOptions` method inside the `Page` or `BrowserContext` base class which you get from `Microsoft.Playwright.NUnit`. See the following example:
```
Comment (dgozman) [inline docs/src/test-runners-csharp.md:113]:
```suggestion
To customize context options, you can override the `ContextOptions` method of your test class derived from `Microsoft.Playwright.NUnit.PageTest` or `Microsoft.Playwright.MSTest.PageTest`. See the following example:
```

File: docs/src/test-runners-csharp.md [hunk 4]
```diff
@@ -108,6 +108,69 @@ By default NUnit will run all test files in parallel, while running tests inside
...
+### Customizing [Browser]/launch options
+
+[Browser]/launch options can be override either using a run settings file or by setting the run settings options directly via the
+CLI. See the following example:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<RunSettings>
+  <TestRunParameters>
+    <Parameter name="browser" value="webkit" />
```
Comment (dgozman) [inline docs/src/test-runners-csharp.md:155]: I don't think `webkit msedge` is going to fly :)

File: docs/src/test-runners-csharp.md [hunk 5]
```diff
@@ -108,6 +108,69 @@ By default NUnit will run all test files in parallel, while running tests inside
...
+    <Parameter name="headless" value="false" />
+    <Parameter name="channel" value="msedge" />
```
Comment (dgozman) [inline docs/src/test-runners-csharp.md:157]: We should list all available options, perhaps in this doc?

File: docs/src/writing-tests-csharp.md [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+    }
+}
+```
+
+</TabItem>
+</Tabs>
+
+### Using Test Hooks
+
+You can use `SetUp` and `TearDown` (or `TestInitialize` and `TestCleanup` in MSTest) as usual to prepare and clean up your test environment:
```
Comment (dgozman) [inline docs/src/writing-tests-csharp.md:169]:
```suggestion
You can use `SetUp`/`TearDown` in NUnit and `TestInitialize`/`TestCleanup` in MSTest to prepare and clean up your test environment:
```

**Review by dgozman — APPROVED** [review]

File: docs/src/test-runners-csharp.md (hunk 5 above)
Comment (mxschmitt) [reply docs/src/test-runners-csharp.md:157]: lets follow up

File: docs/src/browsers.md (hunk 1 above)
Comment (mxschmitt) [reply docs/src/browsers.md:166]: yes, we do.

**Pull request merged**